primitives/src/vrf.rs

```diff
@@ -232,3 +250,155 @@ pub mod sortition {
 		}
 	}
 }
+
+const ENGINE_ID: &[u8; 4] = b"COTS";
+
+pub(super) fn transcript_data(pre_hash: H256, epoch: u64, task_id: H256) -> VRFTranscriptData {
+	VRFTranscriptData {
+		label: ENGINE_ID,
+		items: vec![
+			("epoch", VRFTranscriptValue::U64(epoch)),
+			("task id", VRFTranscriptValue::Bytes(task_id.encode())),
+			("pre hash", VRFTranscriptValue::Bytes(pre_hash.encode())),
+		],
+	}
+}
+
+pub fn prove_vrf(
+	pubkey: PublicKey,
+	pre_hash: H256,
```

Comment:
the last known parent block hash. I've never tried to fetch this value before, there must be a runtime_interface or trait to get it.

Reply:
Ah okay, I see. I was thrown off because in the consensus/PoW code `pre_hash` is the hash of the proposed block header before the seal is attached to it (it's the "pre-seal" hash) and didn't see something analogous here